designer/src/components/TemplateBuilder.tsx

```diff
@@ -0,0 +1,763 @@
+ */
+const convertMustacheTokenToBlock = (token: any): TemplateBlock | null => {
+  const [tokenType, content, ...rest] = token;
+
+  try {
+    switch (tokenType) {
+      case 'text':
+        return {
+          id: Math.random().toString(36).substring(2, 9),
+          type: 'text',
+          content: content,
+        };
+
+      case 'name':
+        return {
+          id: Math.random().toString(36).substring(2, 9),
+          type: 'variable',
+          content: content,
+        };
+
+      case '#':
+        return {
+          id: Math.random().toString(36).substring(2, 9),
+          type: 'if',
+          content: content,
+          children: rest[2].map(convertMustacheTokenToBlock).filter(Boolean),
+        };
+
+      case '^':
+        return {
+          id: Math.random().toString(36).substring(2, 9),
+          type: 'unless',
+          content: content,
+          children: rest[2].map(convertMustacheTokenToBlock).filter(Boolean),
+        };
+
+      default:
+        return null;
+    }
+  } catch (error) {
+    console.warn('Error converting token:', error);
+    return null;
+  }
+};
+
+/**
+ * Type guard for TemplateBlock
+ */
+function isTemplateBlock(block: TemplateBlock | null): block is TemplateBlock {
+  return block !== null;
+}
+
+/**
+ * Component for editing a single template block
+ */
+const TemplateBlockEditor: React.FC<{
+  block: TemplateBlock;
+  variables: Variable[];
+  systemVariables: Variable[];
+  onUpdate: (block: TemplateBlock) => void;
+  onDelete: () => void;
+  onMoveUp?: () => void;
+  onMoveDown?: () => void;
+  isFirst?: boolean;
+  isLast?: boolean;
+  level?: number;
+}> = ({
+  block,
+  variables,
+  systemVariables,
+  onUpdate,
+  onDelete,
+  onMoveUp,
+  onMoveDown,
+  isFirst,
+  isLast,
+  level = 0,
+}) => {
+  const allVariables = [...variables, ...systemVariables];
+
+  const updateBlockContent = (content: string) => {
+    onUpdate({...block, content});
+  };
+
+  const addChildBlock = () => {
+    const newBlock: TemplateBlock = {
+      id: Math.random().toString(36).substring(2, 9),
+      type: 'text',
+      content: '',
+    };
+    onUpdate({
+      ...block,
+      children: [...(block.children || []), newBlock],
+    });
+  };
+
+  return (
+    <Box sx={{pl: 3, borderLeft: '2px solid', borderColor: 'divider', mb: 2}}>
+      <Grid container spacing={2} alignItems="center">
+        <Grid item xs={12} sm="auto">
+          <FormControl fullWidth sx={{minWidth: 120}}>
+            <InputLabel>Block Type</InputLabel>
+            <Select
+              value={block.type}
+              label="Block Type"
+              onChange={e =>
+                onUpdate({...block, type: e.target.value as BlockType})
+              }
+            >
+              <MenuItem value="text">Text</MenuItem>
+              <MenuItem value="variable">Variable</MenuItem>
+              <MenuItem value="if">Conditional</MenuItem>
```

Comment:
Conditional is not clear. When I make a conditional block I choose a variable but then I'm done - i then need to add another block that gets indented... doesn't seem very clear to me.  Maybe just some helper text will do?

Reply:
Also 'Unless' is also a conditional, so maybe 'Conditional' should be 'If'?